### DISCUSSION TITLE
changing parameters

### DISCUSSION THREAD

**Discussion by sandeepmam** [pybamm-team/PyBaMM#1668] [Q&A] [answered]
I am totally a beginner at python. I am trying to change the parameters set which are there in the youtube tutorial, but I am getting errors. where I am doing wrong?

import pybamm
pybamm.Simulation()

Changing parameter sets
model = pybamm.lithium_ion.SPMe()
chemistry = pybamm.parameter_sets.NCA_Kim2011
parameter_values = pybamm.ParameterValues(chemistry=chemistry)
sim = pybamm.Simulation(model,parameter_values=parameter_values)
sim.solve()

---------------------------------------------------------------------------
SolverError                               Traceback (most recent call last)
<ipython-input-27-ba973c6e1a67> in <module>
----> 1 sim.solve()

~\anaconda3\lib\site-packages\pybamm\simulation.py in solve(self, t_eval, solver, check_model, save_at_cycles, starting_solution, **kwargs)
    593             if self.operating_mode == "without experiment":
    594                 if t_eval is None:
--> 595                     raise pybamm.SolverError(
    596                         "'t_eval' must be provided if not using an experiment or "
    597                         "simulating a drive cycle. 't_eval' can be provided as an "

SolverError: 't_eval' must be provided if not using an experiment or simulating a drive cycle. 't_eval' can be provided as an array of times at which to return the solution, or as a list [t0, tf] where t0 is the initial time and tf is the final time. For a constant current (dis)charge the suggested 't_eval'  is [0, 3700/C] where C is the C-rate. For example, run

	sim.solve([0, 3700])

for a 1C discharge.

[change parameter set.txt](https://github.com/pybamm-team/PyBaMM/files/7155564/change.parameter.set.txt)

**Comment by Saransh-cpp** [comment, marked as answer]
> SolverError: 't_eval' must be provided if not using an experiment or simulating a drive cycle. 't_eval' can be provided as an array of times at which to return the solution, or as a list [t0, tf] where t0 is the initial time and tf is the final time. For a constant current (dis)charge the suggested 't_eval' is [0, 3700/C] where C is the C-rate. For example, run

The `solve` method requires `t_eval` with a suggested value of [0, 3700/C] where C is the C-rate.

Replace 
```
sim.solve()
```
with
```
sim.solve([0, 3700])
# or
sim.solve(t_eval=[0, 3700])
```

**Reply by sandeepmam** [reply]
Hi, thanks for the reply.
I have tried but still, I am getting this kind of error.

[change parameter set.txt](https://github.com/pybamm-team/PyBaMM/files/7155821/change.parameter.set.txt)

---------------------------------------------------------------------------
SolverError                               Traceback (most recent call last)
<ipython-input-52-1b3b9795c8a2> in <module>
----> 1 sim.solve([0,3700])

~\anaconda3\lib\site-packages\pybamm\simulation.py in solve(self, t_eval, solver, check_model, save_at_cycles, starting_solution, **kwargs)
    649                         )
    650 
--> 651             self._solution = solver.solve(self.built_model, t_eval, **kwargs)
    652 
    653         elif self.operating_mode == "with experiment":

~\anaconda3\lib\site-packages\pybamm\solvers\base_solver.py in solve(self, model, t_eval, external_variables, inputs, initial_conditions, nproc)
    730             ninputs = len(ext_and_inputs_list)
    731             if ninputs == 1:
--> 732                 new_solution = self._integrate(
    733                     model,
    734                     t_eval_dimensionless[start_index:end_index],

~\anaconda3\lib\site-packages\pybamm\solvers\casadi_solver.py in _integrate(self, model, t_eval, inputs_dict)
    262                                     extrap_event_names.append(event.name[12:])
    263 
--> 264                             raise pybamm.SolverError(
    265                                 "CasADI solver failed because the following "
    266                                 "interpolation bounds were exceeded: {}. You may need "

SolverError: CasADI solver failed because the following interpolation bounds were exceeded: ['nca_ocp_Kim2011_data upper bound']. You may need to provide additional interpolation points outside these bounds.

**Reply by valentinsulzer** [reply]
What did you run? This works on develop
```python
import pybamm

model = pybamm.lithium_ion.SPMe()
chemistry = pybamm.parameter_sets.NCA_Kim2011
parameter_values = pybamm.ParameterValues(chemistry=chemistry)
sim = pybamm.Simulation(model, parameter_values=parameter_values)
sim.solve([0, 3600])
sim.plot()
```

**Reply by sandeepmam** [reply]
I have run this...I would like to get the terminal voltage.

import pybamm

model = pybamm.lithium_ion.SPMe()
chemistry = pybamm.parameter_sets.NCA_Kim2011
parameter_values = pybamm.ParameterValues(chemistry=chemistry)
sim = pybamm.Simulation(model, parameter_values=parameter_values)
sim.solve()
sim.plot(["Terminal voltage[V]"])

**Reply by valentinsulzer** [reply]
Can you update your installation? `pip install git+https://github.com/pybamm-team/pybamm.git@develop`

**Reply by sandeepmam** [reply]
Thank you, I got it.
